File: cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)

{
    // ask user for amount of change
    float n;
    int m;
    do
    {
        n = get_float("How much change is owed to you?: ");
    }
    while (n < 0);

    // convert the dollars into cents and round float to two decimals
    float y = (int)(n * 100 + .5);
    m = y;

    // trickle down check from largest to smallest coin
    // how many quarters will be paid out
    int l = (m % 25);
    int o;
    if (m >= 25)
    {
        o = ((m - l) / 25);
    }
    else
    {
        o = 0;
    }
    // how many nickles will be paid out
    int k = (l % 10);
    int p;
    if (l >= 10)
    {
        p = ((l - k) / 10);
    }
    else
    {
        p = 0;
    }
    // how many dimes will be paid out
    int j = (k % 5);
    int q;
    if (k >= 5)
    {
        q = ((k - j) / 5);
    }
    else
    {
        q = 0;
    }
    // how many pennies will be paid out
    int r;
    if (j >= 1)
    {
        r = j;
    }
    else
    {
        r = 0;
    }
    // total number of coins that will be paid out
    int h = (o + p + q + r);

    // print number of coins used
    printf("%i\n", h);
}

File: sentimental_cash/cash.py
from cs50 import get_float, get_int

# Defining value of coins possible to dispense
QUARTER = 25
DIME = 10
NICKEL = 5
PENNY = 1

# Prompting user for input, which will be stored as float
while True:
    n = get_float("Change owed: ")
    if n > 0:
        break

# Converting float to int
k = int(round(n * 100))

# How many quarters will be dispensed
j = k % QUARTER
if k >= QUARTER:
    o = (k - j) / QUARTER
else:
    o = 0

# How many dimes will be dispensed
i = j % DIME
if j >= DIME:
    p = (j - i) / DIME
else:
    p = 0

# How many nickels will be dispensed
h = i % NICKEL
if i >= NICKEL:
    r = (i - h) / NICKEL
else:
    r = 0

# How many penny will be dispensed
if h >= PENNY:
    s = h
else:
    s = 0

# How many coins will be dispensed
f = o + p + r + s

print(f"{f}")